tests/integration/eventing/userDefinedMessage.ts

```diff
@@ -0,0 +1,162 @@
+
+const RELAY_SERVER_URL = 'http://localhost:3030';
+
+describe('userDefinedMessage', function() {
+  this.timeout(1000 * 60 * 10); // 10 minute timeout for the whole suite
+
+  const createCallTest = async () => {
+    const tokenTtl = 60 * 3; // 3 minute TTL
+
+    const aliceId = `client-id-eventing-tests-alice-${Date.now()}`;
+    const aliceToken = generateAccessToken(aliceId, tokenTtl, env.appSid);
+    const aliceDevice = new Device(aliceToken);
+
+    const bobId = `client-id-eventing-tests-bob-${Date.now()}`;
+    const bobToken = generateAccessToken(bobId, tokenTtl, env.appSid);
+    const bobDevice = new Device(bobToken);
+
+    await bobDevice.register();
+    const bobCallPromise = new Promise<Call>((res) => {
+      bobDevice.on(Device.EventName.Incoming, (c) => res(c));
+    });
+
+    const aliceCall = await aliceDevice.connect({ params: { To: bobId } });
+    const bobCall = await bobCallPromise;
+
+    const aliceCallAcceptPromise = new Promise((res) => {
+      aliceCall.on('accept', () => {
+        res();
+      });
+    });
+    const bobCallAcceptPromise = new Promise((res) => {
+      bobCall.on('accept', () => {
+        res();
+      });
+    });
+
+    bobCall.accept();
+
+    await aliceCallAcceptPromise;
+    await bobCallAcceptPromise;
+
+    const performTeardown = () => {
+      aliceDevice.destroy();
+      bobDevice.destroy();
+    };
+
+    return {
+      performTeardown,
+      alice: {
+        device: aliceDevice,
+        call: aliceCall,
+      },
+      bob: {
+        device: bobDevice,
+        call: bobCall,
+      },
+    };
+  };
+
+  describe('outgoing call leg', function() {
+    it(
+      'should successfully send a message to the customer server',
+      async function() {
+        const { performTeardown, alice: { call } } = await createCallTest();
+
+        const CallSid = call.parameters.CallSid;
+        console.log('call sid', CallSid);
+        const subscription = await axios.post(
+          `${RELAY_SERVER_URL}/create-subscription`,
+          { CallSid },
+        );
+        console.log(`subscription made for call ${CallSid}`, subscription.data);
+
+        const eventSid = call.sendMessage({
+          content: { hello: 'world' },
+          messageType: 'user-defined-message',
+        });
+        await new Promise((res, rej) => {
+          console.log('waiting for twilio to receive the message')
+          call.once('messageSent', (msg) => {
+            if (msg.voiceEventSid === eventSid) {
+              return res();
+            }
+            rej();
+          });
+        });
+        console.log('sent message', eventSid);
+
+        // NOTE(mhuynh): Using a quick solution here such that we just wait for
+        // the message from Twilio to reach the test server. Otherwise, there is
+        // a high likelihood that the message will be missed.
+        await new Promise((res) => {
+          setTimeout(() => {
+            res();
+          }, 3000);
+        });
```

Comment:
use `waitFor`. See connectToken.ts. We should probably move it under util.

Reply:
There's already a `waitFor` in `tests/lib/util.js` and it does something different than the one in `connectToken.ts` :rofl:

Reply:
Do you just want me to copy the one from `connectToken.ts` verbatim?

Reply:
I don't really have a preference. It can be pause, waitFor, etc. As long as the method name is clear.